x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/policy_template_form.tsx

```diff
@@ -119,28 +121,30 @@ const AwsAccountTypeSelect = ({
   updatePolicy: (updatedPolicy: NewPackagePolicy) => void;
   packageInfo: PackageInfo;
 }) => {
-  // This will disable the aws org option for any version LOWER than 1.5.0
+  // This will disable the aws org option for any version below 1.5.0-preview20 which introduced support for account_type. https://github.com/elastic/integrations/pull/6682
   const isValidSemantic = semverValid(packageInfo.version);
   const isAwsOrgDisabled = isValidSemantic
     ? semverCompare(packageInfo.version, AWS_ORG_MINIMUM_PACKAGE_VERSION) < 0
     : true;
 
-  const awsAccountTypeOptions = getAwsAccountTypeOptions(isAwsOrgDisabled);
+  const awsAccountTypeOptions = useMemo(
+    () => getAwsAccountTypeOptions(isAwsOrgDisabled),
+    [isAwsOrgDisabled]
+  );
 
   useEffect(() => {
     if (!getAwsAccountType(input)) {
       updatePolicy(
         getPosturePolicy(newPolicy, input.type, {
           'aws.account_type': {
-            value: awsAccountTypeOptions[0].id,
+            value: isAwsOrgDisabled ? AWS_SINGLE_ACCOUNT : AWS_ORGANIZATION_ACCOUNT,
             type: 'text',
           },
         })
       );
     }
-    // we only wish to call this once on mount
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [input]);
```

Comment:
We need this because we have so many calls to `updatePolicy` when this page mount that each one is overriding the others. I think we should open a tech debt to consolidate all of those calls and create some `getInitialValues` function that will handle all of that logic on its own.

For now this will run on input change but won't pass the if condition unless account type is not set which is only the case on initial mount.